spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/it/TestEntityRepository.java

```diff
@@ -58,6 +58,9 @@
 
 	int countBySize(long size);
 
+	@Nullable
+	List<TestEntity> getByColorAndIdGreaterThanEqualOrderById(String color, Long id);
```

Comment:
Could a test where the id is already a full `Key` with ancestors in a test method too?

Also, I think I might have added in the refdoc that ID properties aren't allowed. We could delete that now.

Reply:
Also in the ref doc I think it should be mentioned that if the user is using `@Query` then he needs to be writing `__key__` himself and that he must use `Key` type params and construct those keys himself even if the properties are long or string.